File: CS/XpoWebApplication/AccentInsensitiveFilterOperator.cs
using DevExpress.Data.Filtering;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace XpoWebApplication {
    public static class AccentInsensitiveFilterStatic {
        public static string Name { get { return "AccentInsensitive"; } }
    }
    public class AccentInsensitiveFilterOperator : ICustomFunctionOperator, ICustomFunctionOperatorFormattable {
        public string Name {
            get { return AccentInsensitiveFilterStatic.Name; }
        }
        string ICustomFunctionOperatorFormattable.Format(Type providerType, params String[] operands) {
            var operand = string.Format("N'%{0}%'", operands[0].Split('\'')[1]);
            return string.Format("{0} COLLATE SQL_Latin1_General_CP1_CI_AI LIKE {1} ", operands[1], operand);
        }
        public Type ResultType(params Type[] operands) {
            return typeof(bool);
        }
        public object Evaluate(params object[] operands) {
            return null;
        }
    }
}

File: VB/XpoWebApplication/AccentInsensitiveFilterOperator.vb
Imports DevExpress.Data.Filtering
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web

Namespace XpoWebApplication
	Public Module AccentInsensitiveFilterStatic
		Public ReadOnly Property Name() As String
			Get
				Return "AccentInsensitive"
			End Get
		End Property
	End Module
	Public Class AccentInsensitiveFilterOperator
		Implements ICustomFunctionOperator, ICustomFunctionOperatorFormattable

		Public ReadOnly Property Name() As String Implements ICustomFunctionOperator.Name
			Get
				Return AccentInsensitiveFilterStatic.Name
			End Get
		End Property
		Private Function ICustomFunctionOperatorFormattable_Format(ByVal providerType As Type, ParamArray ByVal operands() As String) As String Implements ICustomFunctionOperatorFormattable.Format
			Dim operand = String.Format("N'%{0}%'", operands(0).Split("'"c)(1))
			Return String.Format("{0} COLLATE SQL_Latin1_General_CP1_CI_AI LIKE {1} ", operands(1), operand)
		End Function
		Public Function ResultType(ParamArray ByVal operands() As Type) As Type Implements ICustomFunctionOperator.ResultType
			Return GetType(Boolean)
		End Function
		Public Function Evaluate(ParamArray ByVal operands() As Object) As Object Implements ICustomFunctionOperator.Evaluate
			Return Nothing
		End Function
	End Class
End Namespace